pyrevitlib/pyrevit/loader/uimaker.py

```diff
@@ -398,18 +433,341 @@ def _produce_ui_pulldown(ui_maker_params):
     parent_ribbon_panel = ui_maker_params.parent_ui
     pulldown = ui_maker_params.component
 
-    mlogger.debug('Producing pulldown button: %s', pulldown)
+    mlogger.debug("Producing pulldown button: %s", pulldown)
     try:
-        parent_ribbon_panel.create_pulldown_button(pulldown.ui_title,
-                                                   pulldown.icon_file,
-                                                   update_if_exists=True)
+        parent_ribbon_panel.create_pulldown_button(
+            pulldown.ui_title, pulldown.icon_file, update_if_exists=True
+        )
         pulldown_ui = parent_ribbon_panel.ribbon_item(pulldown.ui_title)
 
         _set_highlights(pulldown, pulldown_ui)
 
         return pulldown_ui
     except PyRevitException as err:
-        mlogger.error('UI error: %s', err.msg)
+        mlogger.error("UI error: %s", err.msg)
+        return None
+
+
+def _produce_ui_combobox(ui_maker_params):
```

Comment:
This function is really long, is there a way to split it into logical chunks?
Also, is this code that error prone that you need to catch an exception on each line? (I didn't try it so this is just by reading the code).
If this is necessary, can you reduce the scope of the exceptions to only catch those expected?

Reply:
Alright! I did my best to logically break that down to a few sub-routines to make it easier to read and review now. Still, the complexity is there - mainly because we don't have a single button this time around, but we deal with all the sub-menu (combobox) items as well.